// @jengo/vite@1.0.1/dist/index.js
import { execSync } from 'child_process';
export default function jengo() {
    return {
        name: 'jengo-vite-plugin',
        config: (config, { command }) => {
            if (command === 'build' || command === 'serve') {
                try {
                    const output = execSync('php spark vite:config', { encoding: 'utf-8' });
                    // Find the start of the JSON object
                    const jsonStart = output.indexOf('{');
                    const jsonEnd = output.lastIndexOf('}');
                    if (jsonStart === -1 || jsonEnd === -1) {
                        console.warn('Jengo Vite Plugin: Could not find JSON configuration in output.');
                        return config;
                    }
                    const jsonString = output.substring(jsonStart, jsonEnd + 1);
                    const jengoConfig = JSON.parse(jsonString);
                    if (!jengoConfig || !Array.isArray(jengoConfig.entrypoints)) {
                        console.warn('Jengo Vite Plugin: Invalid configuration format.');
                        return config;
                    }
                    const inputs = {};
                    jengoConfig.entrypoints.forEach((entry) => {
                        // Create a unique key for the entrypoint
                        // e.g. 'app/main.entrypoint.ts' -> 'app_main_entrypoint'
                        // Ensure the path is relative to the root if not already
                        const cleanEntry = entry.startsWith('/') ? entry.substring(1) : entry;
                        const name = cleanEntry.replace(/\//g, '_').replace(/\./g, '_');
                        inputs[name] = cleanEntry;
                    });
                    // console.log(`[Jengo] Found ${Object.keys(inputs).length} entrypoints.`);
                    return {
                        build: {
                            manifest: true,
                            outDir: 'public/dist',
                            rollupOptions: {
                                input: inputs
                            }
                        }
                    };
                }
                catch (error) {
                    console.error('[Jengo] Failed to load configuration:', error);
                }
            }
            return config;
        }
    };
}
